### PR TITLE
cl20: Fix Issue #35 - ask for capability size. Too small size could cause errors

### PR DESCRIPTION
**Pull request by gwawiork**
related with https://github.com/KhronosGroup/OpenCL-CTS/pull/36
Fix Issue #35

### CODE REVIEW

File: test_conformance/conversions/test_conversions.cpp [hunk 1]
```diff
@@ -782,13 +783,20 @@ static int InitCL( void )
...
+    }
+
+    std::vector<char> extensions(set_size);
+    if( (error = clGetDeviceInfo( gDevice, CL_DEVICE_EXTENSIONS, extensions.size(), extensions.data(),  NULL ) ) )
     {
         vlog_error( "FAILURE: unable to get device info for CL_DEVICE_EXTENSIONS!" );
         return -1;
     }
-    else if( strstr( extensions, "cl_khr_fp64" ) )
+    else if( strstr( extensions.data(), "cl_khr_fp64" ) )
```
Comment (kpet) [inline test_conformance/conversions/test_conversions.cpp]: `is_extension_available`? I don't think performance is a concern here.

File: test_conformance/conversions/test_conversions.cpp [hunk 2]
```diff
@@ -800,7 +808,7 @@ static int InitCL( void )
     else if( strstr(profile, "EMBEDDED_PROFILE" ) )
     {
         gIsEmbedded = 1;
-        if( !strstr( extensions, "cles_khr_int64" ) )
+        if( !strstr( extensions.data(), "cles_khr_int64" ) )
```
Comment (kpet) [inline test_conformance/conversions/test_conversions.cpp]: Same here.

**Comment by gwawiork** [comment]
Ok I will open against master again. 
Regrading cl12_trunk - I see many differences comparing to other branches (e.g conversions tests). Then I will skip this.

**Comment by kpet** [comment]
Thanks. Yes cl12_trunk is a separate problem. We should try to port changes but only when the effort required is reasonable.

**Pull request merged**